src/api/controllers/adminPermissionsController.js

```diff
@@ -0,0 +1,165 @@
+import models from '../models';
+import statusCodes from '../../helpers/constants/status.codes';
+import generatePagination from '../../helpers/generate.pagination.details';
+
+const {
+  Article, Report, User, Reason
+} = models;
+
+/**
+ *
+ *
+ * this is the article controller
+ *
+ * @export
+ * @class ArticleController
+ */
+
+/**
+ * containing all article's model controllers
+ * @export
+ * @class ArticleController
+ */
+export default class adminPermissionsController {
+  /**
+   * fetch all reported Articles
+   *
+   * @static
+   * @param {array} req - request array
+   * @param {array} res - respond array
+   * @returns {array} response body
+   * @memberof getArticles
+   */
+  static async getReportedArticles(req, res) {
+    const defaultOffset = 0;
+    const defaultLimit = 10;
+    const offset = req.query.offset || defaultOffset;
+    const limit = req.query.limit || defaultLimit;
+    const paginatedArticle = await Report.findAll({
+      attributes: [['createdAt', 'ReportedOn']],
+      include: [
+        {
+          model: Article,
+          required: true,
+          attributes: ['title', 'slug', 'description', 'body', 'coverImage']
+        },
+        {
+          model: User,
+          required: true,
+          attributes: ['username', 'email']
+        },
+        {
+          model: Reason,
+          required: true,
+          attributes: ['description']
+        }
+      ],
+      offset,
+      limit
+    });
+    res.status(statusCodes.OK).send({
+      status: statusCodes.OK,
+      paginationDetails: generatePagination(paginatedArticle.length,
+        paginatedArticle,
+        offset,
+        limit),
+      data: paginatedArticle
+    });
+  }
+
+  /**
+   * fetch all reported Articles
```

Comment:
is it all reported articles or a single article

Reply:
I resolved this